test/plugin/example_plugin_test.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"os"
+	"testing"
+	"time"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func ReadNLines(reader io.Reader, n int) ([]string, error) {
+	var lines []string
+	reReader := bufio.NewReader(reader)
+	for i := 0; i < n; i++ {
+		line, err := reReader.ReadString('\n')
+		lines = append(lines, line)
+		if err != nil {
+			return nil, fmt.Errorf("failed reading %d-th line: %s", i, err)
+		}
+	}
+
+	return lines, nil
+}
+func TestTCPPlugin(t *testing.T) {
+	host := os.Getenv("SECRETLESS_HOST")
+	if host == "" {
+		host = "localhost"
+	}
+
+	address := host + ":6175"
+
+	log.Printf("Trying to use host: %s", address)
+
+	Convey("can inject information from credentials and proxy connection", t, func() {
+
+		// initial write
+		connection, err := net.Dial("tcp", address)
+		So(err, ShouldBeNil)
+		defer connection.Close()
+		connection.SetWriteDeadline(time.Now().Add(3 * time.Second))
+		connection.SetReadDeadline(time.Now().Add(3 * time.Second))
```

Comment:
This should probably be a constant up top

Reply:
Why are we using `Convey` in new code?

Reply:
> This should probably be a constant up top

should it?  not sure there's any benefit to the abstraction here and it might actually harm readability.

Reply:
It's useful to break this test into parts. I wonder what y'all think. In this way the connection sort of becomes a constant (if that's what was being talked about). If it was the deadlines then those have been refactored into something even better.

Also the connector interface definitely needs the deadline setting capabilities of net.Conn vs io.ReaderWriter. Quite a bit relies on being able to set deadlines